jsonapi/marshal.go

```diff
@@ -285,67 +287,68 @@ func getStructRelationships(relationer MarshalLinkedRelations, information Serve
 		relationships[name] = relationship
 	}
 
-	return &relationships
+	return relationships
 }
 
 // helper method to generate URL fields for `links`
 func getLinksForServerInformation(relationer MarshalLinkedRelations, name string, information ServerInformation) *Links {
-	links := &Links{}
-
-	if information != serverInformationNil {
-		prefix := strings.Trim(information.GetBaseURL(), "/")
-		namespace := strings.Trim(information.GetPrefix(), "/")
-		structType := getStructType(relationer)
+	if information == nil {
```

Comment:
I can only find one situation:
https://github.com/manyminds/api2go/blob/5207f05d44090999ebf2e09b3e66566484c4dbee/api.go#L227

It might be possible that `info = &api.info` can return one such a nil typed interface if it is not set?